File: storage/data_transpiled/at_coder/abc121/B/4932719.py
def _import ( ) : return _import ( )
import math
class Main ( object ) :
    def __init__ ( self ) :
        self.n = 0
        self.m = 1
        self.ni = 0
        self.hi = 0
        self.wi = 0
        self.lefi = 0
        self.rigi = 0
        self.ansi = 0
        self.s = ''
        self.t = ''
        self.anss = ''
        self.sAr = ''
        self.ansl = ''
        self.ari = ''
        self.sqari = ''
        self.w = ''
        self.ww = ''
        self.isvisited = True
    def main ( self ) :
        sc = sys.stdin
        self.n = self.n
        self.m = self.m
        self.c = self.c * - 1
        self.ansi = 0
        bset = [ ]
        asets = [ ]
        for i in range ( self.m ) :
            bset.append ( sc.read ( ) )
        for i in range ( self.n ) :
            for j in range ( self.m ) :
                asets [ i ].append ( sc.read ( ) )
        for i in range ( self.n ) :
            tmp = 0
            for j in range ( self.m ) :
                tmp += bset [ j ] * asets [ i ] [ j ]
            if tmp > self.c :
                self.ansi += 1
            tmp = 0
        print ( self.ansi )
        sc.close ( )
    def gcd_int ( self , a , b ) :
        if self.a < b :
            tmpp = self.a
            self.a = b
            self.b = tmpp
        if self.b == 0 :
            return a
        else :
            return gcd_int ( self , a % b )
    def gcd_long ( self , a , b ) :
        if self.a < b :
            tmpp = self.a
            self.b = tmpp
        if self.b == 0 :
            return a
        else :
            return gcd_long ( self , a % b )

File: data/data_original/at_coder/abc121/B/4932719.java
/*
* @Author Silviase(@silviasetitech)
* For ProCon
*/

import java.util.*;
import java.lang.*;
import java.math.*;

class Main{

        // usual use input int
        static int n;
        static int m;
        static int ni;
        static int hi;
        static int wi;
        static int lefi;
        static int rigi;
        static int ansi;
        //
        static String s;
        static String t;
        static String anss;
        // 
        static String[] sAr;
        //
        static long ansl;
        //
        static int[] ari;
        //
        static int[][] sqari;
        //

        // for dfs
        static int[] w;
        static int[] ww;
        static boolean[] isvisited;
        //
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
            
        int n = sc.nextInt();
        int m = sc.nextInt();
        int c = sc.nextInt();
        c = c*-1;
        ansi = 0;
        int[] bset = new int[m];
        int[][] asets = new int[n][m];

        for (int i = 0; i < m; i++) {
            bset[i] = sc.nextInt();
        }
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                asets[i][j] = sc.nextInt();
            }
        }

        for (int i = 0; i < n; i++) {
            int tmp = 0;
            for (int j = 0; j < m; j++) {
                tmp += bset[j]*asets[i][j];
            }
            if (tmp > c) {
                ansi++;
            }
            tmp = 0;
        }
        System.out.println(ansi);


        sc.close();
    }


        public static int gcd_int(int a, int b) {
        
            if (a < b) {
                int tmpp = a;
                a = b;
                b = tmpp;
            }
            if (b == 0) {
                return a;
            }else{
                return gcd_int(b, a%b);
            }
        }

        public static long gcd_long(long a, long b) {
        
            if (a < b) {
                long tmpp = a;
                a = b;
                b = tmpp;
            }
            if (b == 0) {
                return a;
            }else{
                return gcd_long(b, a%b);
            }
        }

        public static int lcm_int(int a, int b) {
            int gcd = gcd_int(a,b);
            return a/gcd*b;
        }

        public static long lcm_long(long a, long b) {
            long gcd = gcd_long(a,b);
            return a/gcd*b;
        }

        public static void dfs(int placenow) {
            // if went all -> success! 
            // if not went all -> fail...
            
            /*
            dfs    
            Go All Place that there is way to and not having been yet. 

             if island 1 is start point, dfs(1);
            if there is way to island 2 and island 3,
            - island 2 changes non-visited -> visited, and dfs(2).
            - island 3 changes non-visited -> visited, and dfs(3).
    
            */

            isvisited[placenow] = true;
            boolean success = true;
            for (int i = 0; i < n; i++) {
                if (isvisited[i] == false) { // not go all place
                    success = false;
                    break;
                }
            }
            if (success) {
                ansi++;
                isvisited[placenow] = false;
                return;
            }

            for (int i = 0; i < n; i++) {
                if (w[i] == placenow && isvisited[ww[i]] == false ) {
                    dfs(ww[i]);
                }else if(ww[i] == placenow && isvisited[w[i]] == false){
                    dfs(w[i]);
                }else{
                    continue;
                }
            }
            isvisited[placenow] = false;
            return;
        }

}